sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java

```diff
@@ -0,0 +1,127 @@
+package io.sentry.android.core;
+
+import io.sentry.IPerformanceContinuousCollector;
+import io.sentry.ISpan;
+import io.sentry.NoOpSpan;
+import io.sentry.NoOpTransaction;
+import io.sentry.SpanDataConvention;
+import io.sentry.SpanId;
+import io.sentry.android.core.internal.util.SentryFrameMetricsCollector;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public class SpanFrameMetricsCollector
+    implements IPerformanceContinuousCollector,
+        SentryFrameMetricsCollector.FrameMetricsCollectorListener {
+  private @NotNull final Object lock = new Object();
+  private @Nullable final SentryFrameMetricsCollector frameMetricsCollector;
+  private @Nullable volatile String listenerId;
+  private @NotNull final Map<SpanId, FrameMetrics> metricsAtSpanStart;
+
+  private @NotNull final FrameMetrics currentFrameMetrics;
+  float lastRefreshRate = 0;
```

Comment:
I guess this should be private, but also didn't see where it's used - is this gonna be in another PR?

Reply:
Oh true, we can remove this within this PR - thanks!